src/main/webapp/wise5/components/outsideURL/outside-url-authoring/outside-url-authoring.component.ts

```diff
@@ -0,0 +1,146 @@
+    super.ngOnInit();
+    this.outsideURLIFrameId = 'outsideResource_' + this.componentId;
+    this.isShowOERs = this.componentContent.url === '';
+    this.subjects = [
+      {
+        value: 'Earth and Space Sciences',
+        label: $localize`Earth and Space Sciences`
+      },
+      {
+        value: 'Life Sciences',
+        label: $localize`Life Sciences`
+      },
+      {
+        value: 'Physical Sciences',
+        label: $localize`Physical Sciences`
+      },
+      {
+        value: 'Engineering, Technology, and Applications of Science',
+        label: $localize`Engineering, Technology, and Applications of Science`
+      }
+    ];
+    this.searchText = '';
+    this.selectedSubjects = [];
+    this.OutsideURLService.getOpenEducationalResources().then((openEducationalResources: any) => {
+      this.allOpenEducationalResources = openEducationalResources.sort((a, b) =>
+        a.metadata.title > b.metadata.title ? 1 : -1
+      );
+      this.filteredOpenEducationalResources = this.allOpenEducationalResources;
+    });
+    this.urlChangedSubscription = this.urlChanged
+      .pipe(debounceTime(1000), distinctUntilChanged())
+      .subscribe((url: string) => {
+        this.authoringComponentContent.url = url;
+        this.authoringComponentContent.info = null;
+        this.componentChanged();
+      });
+    this.widthChangedSubscription = this.widthChanged
+      .pipe(debounceTime(1000), distinctUntilChanged())
+      .subscribe((width: string) => {
+        this.componentChanged();
+      });
+    this.heightChangedSubscription = this.heightChanged
+      .pipe(debounceTime(1000), distinctUntilChanged())
+      .subscribe((height: string) => {
+        this.componentChanged();
+      });
+  }
+
+  ngOnDestroy() {
+    this.urlChangedSubscription.unsubscribe();
+    this.widthChangedSubscription.unsubscribe();
+    this.heightChangedSubscription.unsubscribe();
+  }
+
+  chooseOpenEducationalResource(openEducationalResource: any): void {
+    this.authoringComponentContent.url = openEducationalResource.url;
+    this.authoringComponentContent.info = openEducationalResource.info;
+    this.componentChanged();
+  }
+
+  isResourceSelected(resourceUrl: string): boolean {
+    return resourceUrl === this.authoringComponentContent.url;
+  }
+
+  reloadResource(): void {
+    const iframe: any = document.getElementById(this.outsideURLIFrameId);
+    iframe.src = '';
+    iframe.src = this.authoringComponentContent.url;
+  }
+
+  clearFilters(): void {
+    this.searchText = '';
+    this.selectedSubjects = [];
+    this.searchFieldChanged();
+  }
+
+  searchFieldChanged(): void {
+    this.filteredOpenEducationalResources = this.allOpenEducationalResources.filter((oer) => {
+      if (!this.isTextMatch(this.searchText, JSON.stringify(oer))) {
+        return false;
+      }
+      if (this.isAnySubjectChosen()) {
+        return this.isSubjectMatch(this.selectedSubjects, oer);
+      } else {
+        return true;
+      }
```

Comment:
This logic was a bit difficult to follow for me because of the negation and the order in which it's testing things. What do you think about re-writing it like this?
```suggestion
      if (this.isAnySubjectChosen()) {
        return (
          this.isSubjectMatch(this.selectedSubjects, oer) &&
          this.isTextMatch(this.searchText, JSON.stringify(oer))
        );
      }
      return this.isTextMatch(this.searchText, JSON.stringify(oer));
```